extensions/amp-analytics/0.1/variables.js

```diff
@@ -0,0 +1,236 @@
+    this.register_('trim', value => value.trim());
+    this.register_('json', value => JSON.stringify(value));
+    this.register_('toLowerCase', value => value.toLowerCase());
+    this.register_('toUpperCase', value => value.toUpperCase());
+    this.register_('not', value => String(!value));
+    this.register_('base64', value => base64Encode(String(value)));
+    this.register_('if', (value, thenValue, elseValue) => Boolean(value)
+      ? thenValue
+      : elseValue);
+
+    cryptoFor(this.win_).then(crypto => {
+      this.crypto_ = crypto;
+    });
+  }
+
+  /**
+   * [register_ description]
+   * @param  {string} name [description]
+   * @param  {VariableFilterDef} handler [description]
+   */
+  register_(name, handler) {
+    if (this.filters_[name]) {
+      dev().error(TAG, 'Filter "' + name + '" already registered.');
+    }
+    dev().assert(handler, 'Handler for filter ' + name + ' is invalid.');
+    this.filters_[name] = handler;
+  }
+
+
+  /**
+   * [parseFilter_ description]
+   * @param  {string} filterStr [description]
+   * @return {!Object<string, *>}  [description]
+   */
+  parseFilter_(filterStr) {
+    if (!filterStr) {
+      return {};
+    }
+
+    // TODO (avimehta): The parsing for filters breaks when `:` is used as
+    // something other than the argument separator. A full-fledged parser would
+    // be needed to fix this.
+    const tokens = filterStr.split(':');
+    user().assert(tokens[0], 'Filter ' + name + ' is invalid.');
+    const fn = this.filters_[tokens[0]];
+    if (!fn) {
+      user().error(TAG, 'Invalid filter name: ' + tokens[0]);
+      return {};
+    }
+    return {fn, args: tokens.splice(1)};
+  }
+
+  /**
+   * [applyFilters_ description]
+   * @param  {VariableFilterIODef} value   [description]
+   * @param  {Array<string>} filters [description]
+   * @return {VariableFilterIODef}  [description]
+   */
+  applyFilters_(value, filters) {
+    for (let i = 0; i < filters.length; i++) {
+      const parsedFilter = this.parseFilter_(filters[i].trim());
+      if (parsedFilter.fn) {
+        parsedFilter.args.splice(0, 0, value);
+        value = parsedFilter.fn.apply(null, parsedFilter.args);
+      }
+    }
+    return value;
+  }
+
+  /**
+   * @param {string} template The template to expand.
+   * @param {!JSONType} trigger The object to use for variable value lookups.
+   * @param {!Object=} opt_event Object with details about the event.
+   * @param {number=} opt_iterations Number of recursive expansions to perform.
+   *    Defaults to 2 substitutions.
+   * @param {boolean=} opt_encode Used to determine if the vars should be
+   *    encoded or not. Defaults to true.
+   * @return {string} The expanded string.
+   */
+  expandTemplate(template, trigger, config, opt_event, opt_iterations,
+      opt_encode) {
+    opt_iterations = opt_iterations === undefined ? 2 : opt_iterations;
+    opt_encode = opt_encode === undefined ? true : opt_encode;
```

Comment:
Calling this `opt_encode` make me think the default is to _not_ encode. Can we call this `opt_no_encode`, which its default value being `false` (you'll have to swap any `falses`you passed before to `true`).